account.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"math"
+	"time"
+
+	"github.com/cockroachdb/apd/v2"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// Account is an internal representation of a genesis regen account
+type Account struct {
+	Address       sdk.AccAddress
+	TotalRegen    apd.Decimal
+	StartTime     time.Time
+	Distributions []Distribution
+	EndTime       time.Time
+}
+
+// Distribution is an internal representation of a genesis vesting distribution of regen
+type Distribution struct {
+	Time  time.Time
+	Regen apd.Decimal
+}
+
+const (
+	URegenDenom = "uregen"
+)
+
+var tenE6 *apd.Decimal
+
+func init() {
+	var err error
+	tenE6, _, err = apd.NewFromString("1000000")
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ToCosmosAccount(acc Account) (auth.AccountI, *bank.Balance, error) {
+	totalCoins, err := RegenToCoins(&acc.TotalRegen)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	addrStr := acc.Address.String()
+	balance := &bank.Balance{
+		Address: addrStr,
+		Coins:   totalCoins,
+	}
+	if len(acc.Distributions) == 0 {
+		return &auth.BaseAccount{Address: addrStr}, balance, nil
+	} else {
+		var periods []vesting.Period
+
+		periodStart := acc.StartTime
+
+		for _, dist := range acc.Distributions {
+			coins, err := RegenToCoins(&dist.Regen)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			length := dist.Time.Sub(periodStart)
```

Comment:
I think this is actually not the desired behavior.

`length` should always be the duration of time between previous distribution and the current distribution's releasing of funds (see [vesting implementation](https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/vesting/types/vesting_account.go#L396-L407)).

Since we want the first trench to vest immediately, the first period.length should always be 0, and each period thereafter should be calculated as: `dist.Time.Sub(prevDist.Time)`. I think we could potentially even hard code them as 1 month for all `length` values, but I do think it's actually more accurate to derive them from the `dist.Time` values.

Reply:
I think the error is that I'm not updating `periodStart = dist.Time` after this. With this logic, the first period will always be 0. 1 month is dealt with elsewhere and shouldn't be here because of the merge logic